### PR TITLE
Add missing clocks

### PR DESCRIPTION
**Pull request by hmvp**
Also reorderd the macro invocations to match the datasheet

### CODE REVIEW

File: rp2040-hal/src/clocks/mod.rs [hunk 1]
```diff
@@ -101,6 +150,30 @@ pub trait ClockSYSClockAuxSource {
...
     fn set_clksys_auxsrc(&mut self);
 }
+
+/// For clocks that can have the USB Clock as an auxilliary source.
+pub trait ClockUSBClockAuxSource {
+    /// Set USB Clock as source.
+    fn set_clkusb_auxsrc(&mut self);
+}
+
+/// For clocks that can have the USB Clock as an auxilliary source.
```
Comment (9names) [inline rp2040-hal/src/clocks/mod.rs:160]: Copy paste error on the comment for the ADC, RTC, Ref clock blocks

**Review by 9names — APPROVED** [review]
Looks good to me. A couple of comment errors due to copy+paste, but other than that this looks good to me.
If you want to fix these go ahead, otherwise we'll do a cleanup post-merge.
Thanks for the PR

File: rp2040-hal/src/clocks/mod.rs (hunk 1 above)
Comment (hmvp) [reply rp2040-hal/src/clocks/mod.rs:160]: no problem, i will clean them up in a moment

Comment (hmvp) [reply rp2040-hal/src/clocks/mod.rs]: Fixed

**Pull request merged**